scapy/contrib/automotive/uds_scan.py

```diff
@@ -320,6 +322,183 @@ def _get_table_entry(tup):
                 label)
 
 
+class UDS_RDBISelectiveEnumerator(StagedAutomotiveTestCase):
+    @staticmethod
+    def __connector_random_to_sequential(rdbi_random, _):
+        # The scan range for the sequential enumerator
+        # is generated in the sequential enumerator itself,
+        # since in this connector we do not know the state
+        # of the ECU.
+        identifiers_with_positive_response = \
+            [p.resp.dataIdentifier
+             for p in rdbi_random.results_with_positive_response]
+
+        scan_range = UDS_RDBISelectiveEnumerator.\
+            _random_results_to_scan_range(identifiers_with_positive_response)
+        return {"scan_range": scan_range}
+
+    @staticmethod
+    def _poi_to_scan_range(pois):
+        # type: (UDS_RDBIRandomEnumerator) -> Iterable[int]
+
+        if len(pois) == 0:
+            # quick path for better performance
+            scan_range = []
+        else:
+            block_size = UDS_RDBIRandomEnumerator.block_size
+            to_scan = []
+            for block_index, start in enumerate(range(0, 2 ** 16, block_size)):
+                end = start + block_size - 1
+                pr_in_block = any((start <= identifier <= end
+                                   for identifier in pois))
+                if pr_in_block:
+                    to_scan += range(start, end + 1)
+            # make all identifiers unique with set()
+            # Sort for better logs
+            scan_range = sorted(list(set(to_scan)))
```

Comment:
I would leave the scan_range es generator here... so you can do the following,
```return itertools.chain.from_iterable([range(1,5), range(1,9)])```

But this only works, if a generator is sufficient for the next stage. Maybe you need to pass that as a `list`

Reply:
Refactored